crates/libs/bindgen/src/rust/handles.rs

```diff
@@ -63,12 +63,47 @@ pub fn gen_win_handle(writer: &Writer, def: metadata::TypeDef) -> TokenStream {
         }
     };
 
+    let free = if let Some(function) = free_function(def) {
+        if is_invalid.is_empty() {
+            // TODO: https://github.com/microsoft/win32metadata/issues/1891
+            quote! {}
+        } else {
+            let name = to_ident(function.name());
+            let signature = metadata::method_def_signature(def.namespace(), function, &[]);
+
+            // BCryptCloseAlgorithmProvider has an unused trailing parameter.
+            let tail = if signature.params.len() > 1 {
+                quote! { , 0 }
+            } else {
+                quote! {}
+            };
```

Comment:
Just heads up, I just added metadata for AMSI context/session freeing and this will likely get tangled up here; `AmsiCloseSession` needs two valid inputs 🙃

Perhaps we should mark `BCryptCloseAlgorithmProvider::dwFlags` as `Reserved` then you can key off that and decide whether to abort?

Reply:
I can limit my additions to those that fit this form for now. Might be better to then have that discussion on the metadata side of the fence.